code/modules/mob/living/carbon/xenomorph/mutators/strains/drone/healer.dm

```diff
@@ -173,3 +183,122 @@
 /datum/behavior_delegate/drone_healer/proc/un_salve()
 	salve_applied_recently = FALSE
 	bound_xeno.update_icons()
+
+/*
+	SACRIFICE
+*/
+
+/datum/behavior_delegate/drone_healer/proc/modify_transferred(amount)
+	transferred_amount += amount
+	if(transferred_amount > max_transferred_amount)
+		transferred_amount = max_transferred_amount
+	if(transferred_amount < 0)
+		transferred_amount = 0
+
+/datum/behavior_delegate/drone_healer/append_to_stat()
+	. = list()
+	. += "Transferred health amount: [transferred_amount]"
+	if(transferred_amount >= max_transferred_amount)
+		. += "Sacrifice will grant you new life."
+
+/datum/behavior_delegate/drone_healer/on_life()
+	if(!bound_xeno)
+		return
+	if(bound_xeno.stat == DEAD)
+		return
+	var/image/holder = bound_xeno.hud_list[PLASMA_HUD]
+	holder.overlays.Cut()
+	var/percentage_transferred = round((transferred_amount / max_transferred_amount) * 100, 10)
+	if(percentage_transferred)
+		holder.overlays += image('icons/mob/hud/hud.dmi', "xenoenergy[percentage_transferred]")
```

Comment:
Thinking about moving this into a separate proc that triggers on modify_transferred(). But not sure if that would be better, because you can heal many times in two seconds and updating HUD every time sounds like a waste + inconsistent.